### ISSUE TITLE
Accessibility: unlabeled buttons in appointment screen

### ISSUE THREAD

**Issue by ndarilek** [loconomics/loconomics#660]
From my calendar, I clicked a link labeled "
12:00am12:00amUnavailable". This brought me to a page with a URL of the form:

http://localhost:8811/appDebug.html#!appointment/2017-09-28T05:00:00.000Z/-5

On this screen, there are two unlabeled buttons below the heading containing the date.

As an aside, I assume this is a New Appointment page of some sort, but the link text isn't clear to me.

**Comment by IagoSRL** [comment]
Thanks @ndarilek, good catch. Joshua and me were reviewing this and we will do some changes:
- at calendar:
  - we will move the link wrapping all the text (like "12:00am12:00amUnavailable") to a button after that/on the right with a correct label
  - for cases like this that is 'unavailable' should not be a link, is just not useful and confusing
  - for 'free' availability, at first will not be a link but later we will put a link that allows user to choose to 'add booking' or 'add calendar block',
  - for actual events, the link will say 'see' and go to display a card with full details, and depending on the case the card can have an 'edit' button (if is editable).
- at appointment: we have a pending refactoring of this; reviewing it we found will be much easier if we replaced this with something that displays just the card with event details without the navigation elements, just a 'back' button to return to the calendar.

We think will be easier for all users (and too the implementation ;-), but let us know your opinion or any advice.

**Comment by IagoSRL** [comment]
I quickly removed navigation elements from the appointment (only html, still need further refactoring) so must prevent some unusable and unaccessible states.

**Comment by joshdanielson** [comment]
@ndarilek I made lots of revisions to the appointments/events shown in the calendar and added custom aria-labels for each. Please test them here: http://localhost:8811/appDebug.html#!calendar/2017-10-03T07:00:00.000Z.

**Comment by ndarilek** [comment]
Cool, now when I visit that screen, I'm told that I'm free all day. I'm 
not sure if there are other activities I'm supposed to be able to 
perform there, but it's certainly less confusing.

**Comment by joshdanielson** [comment]
@ndarilek did the "Free" time slot say "Free from 9am to 5pm. Click to add a new booking or calendar block."? the times may be different but wanted to check the rest.

**Comment by ndarilek** [comment]
My results on this may have been invalid, `grunt atwork` may not have 
been compiling the latest code to my build/ directory. Apologies, will 
retest once I figure out how to get things back in order.

**Comment by ndarilek** [comment]
On commit 63cdc7f, after a fresh `grunt build`, I still hear "12:00 
AM-11:59 PM: Free". I do hear "Click to change date" when highlighting 
the button titled with today's date, so the display is different in that 
respect.

**Comment by joshdanielson** [comment]
@ndarilek let's try you making a test booking so another event will show up in your calendar to test and we should be testing the booking anyway. Can you make a booking with me using this link? http://localhost:8811/appDebug.html#!/appDebug.html/booking/540/106

For the credit card, just enter 4111111111111111 with 10/2018 for expiration, 045 for the ccv, and 94110 for the zip code. 

Any questions, let me know, as there may be issues we don't know about yet (hopefully not).

**Comment by ndarilek** [comment]
Notes on the booking process:

  * When advancing to another step, it would be helpful for focus to 
land on the first control of that screen. Certainly not a deal-breaker, 
but definitely a nice-to-have.
  * When I select a date, it would be useful to pop focus to, say, the 
first available time on that date. Other possibilities are the date's 
<h2> or the paragraph below listing the time requirements.
  * I thought I'd need an hour 15, but on the confirmation screen I read:
"
Wednesday, October 4th, 10:00am-1:15pm"
  * On the confirmation screen, many apparent controls aren't in tab 
focus. I can tab directly from "Your instant booking, step 5 of 5" to 
"Confirm". There appear to be several buttons between these steps that I 
can't tab to. Likewise, a control labeled "Add notes to Joshua" isn't 
tabbable nor does it indicate what it is (I.e. button, link, etc.)

Having booked an appointment for tomorrow and visiting:

http://localhost:8811/appDebug.html#!appointment/2017-10-04T05:00:00.000Z/-2

I don't see it listed and I'm told that I'm completely free. Am I doing 
something wrong?

**Comment by joshdanielson** [comment]
@ndarilek great notes, let me review them with @IagoSRL. 

Try this view: http://localhost:8811/appDebug.html#!calendar/2017-10-04T07:00:00.000Z

It may be that you're viewing the appointment card, and we decided to wait to make that fully accessible until we do a refactor (coming in the next few weeks) so apologies for having you test that.

**Comment by ndarilek** [comment]
http://localhost:8811/appDebug.html#!calendar/2017-10-04T05:00:00.000Z 
still shows me as free.

I'll forward my appointment confirmation email to you in case that has 
any hints as to why this isn't showing up for me.

**Comment by joshdanielson** [comment]
@ndarilek I"m getting the same problem when I view my calendar after booking another professional. @IagoSRL I'm not sure why it wouldn't show bookings for other professionals, is this a bug?
